review section 5: Interoperability with Jakarta EE specs (#352)
jakartaee/data: spec/src/main/asciidoc/jakarta-ee.adoc

=== Provider Name

The entity annotation class is the primary strategy to avoid conflicts between Jakarta Data providers. In most cases, it is sufficient to differentiate between providers. In situations where multiple Jakarta Data providers support the same entity annotation class, the application can specify the name of the desired Jakarta Data provider using the optional `provider` attribute of the `Repository` annotation.

To ensure compatibility and prevent conflicts, Jakarta Data providers must disregard all `Repository` annotations that specify a different provider's name through the `Repository.provider()` attribute. By ignoring these annotations, Jakarta Data providers allow other Jakarta Data providers to handle them.

== Interoperability with other Jakarta EE Specifications

This section discusses interoperability with related Jakarta EE specifications. When operating within a Jakarta EE product, the availability of other Jakarta EE technologies depends on whether the Jakarta EE Core profile, Jakarta EE Web profile, or Jakarta EE Platform is used.

=== Jakarta Contexts and Dependency Injection

> Contexts and Dependency Injection (CDI) is a specification in the Jakarta EE Core profile that provides a powerful and flexible dependency injection framework for Java applications. CDI lets you decouple components and manage their lifecycle through dependency injection, enabling loose coupling and promoting modular and reusable code.

Reviewer: ```suggestion
```
I don't think we need to define CDI, though I could do this in a separate pull again, if we want one specifically for making the spec more concise.

> In Jakarta EE products, CDI provides the ability to register implementation of Jakarta Data repositories for injection into applications via the the `@Inject` annotation. Repositories are interfaces that define data access and manipulation operations for a specific domain entity. The following example illustrates this integration:

Reviewer: ```suggestion
In Jakarta EE products, CDI provides the ability to register implementations of Jakarta Data repositories for injection into applications via the the `@Inject` annotation.  The following example illustrates this integration:
```
This doesn't seem like the right point to define Repositories

Author: I agree. It seemed out of place to me as well.  I'll add your suggestion to remove it.

}
----

> In the above example, a `CarRepository` interface extends the `BasicRepository` interface provided by Jakarta EE Data. The `BasicRepository` interface offers a set of basic operations for entities.

Reviewer: ```suggestion
In the above example, a `CarRepository` interface extends the `BasicRepository` interface provided by Jakarta Data. The `BasicRepository` interface offers a set of basic operations for entities.
```

> By annotating the `CarRepository` interface with `@Repository`, the application requests that the Jakarta Data provider generate implementation of the interface and its methods. To make the implementation available to the application, the Jakarta Data provider uses CDI to register the implementation as a bean or registers a producer or producer factory to provide the implementation on unqualified injection points for the `CarRepository` interface.

Reviewer: ```suggestion
By annotating the `CarRepository` interface with `@Repository`, the application requests that the Jakarta Data provider generate an implementation of the interface and its methods. To make the implementation available to the application, the Jakarta Data provider uses CDI to register the implementation as a bean, or registers a producer or producer factory to provide the implementation on unqualified injection points for the `CarRepository` interface.
```

Author: The first part of the suggested change (inserting "an") looks correct, but the second part to add a comma looks incorrect because it is between two verbs ("uses" and "registers") for the same subject ("the Jakarta Data provider").

It would be correct to add the comma if we also add the word "it" so that it becomes a compound sentence,
> ... the Jakarta Data provider uses CDI to register the implementation as a bean, or it registers ...

Reviewer: Yep, you're right that's a compound predicate which would be wrong to have a comma, I'll update it.

Author: Thanks, I'll add it now.

// ...

List<Car> cars = repository.findByType(CarType.SPORT);
----

In the above snippet, the application injects the `CarRepository` instance using the `@Inject` annotation. Once injected, the repository object can be used to invoke various data access methods defined by the `CarRepository` interface, such as `save()`, `findByType()`, and `findByName()`.

> This integration between CDI and Jakarta EE Data allows for seamless management of repository instances within Jakarta EE applications.

Reviewer: ```suggestion
This integration between CDI and Jakarta Data allows for seamless management of repository instances within Jakarta EE applications.
```
